src/components/App/index.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import './styles.css';
+
+import Display from '../Display';
+import ButtonPanel from '../ButtonPanel';
+
+import calculate from '../../services/calculate';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+      calculated: false,
+    }
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const result = calculate(this.state, buttonName);
```

Comment:
Since we are using the (soon to be previous) state here, a better way we can do this is:

```javascript
handleClick = (buttonName) => this.setState(prevState => calculate(prevState, buttonName));
```
[This](https://medium.freecodecamp.org/functional-setstate-is-the-future-of-react-374f30401b6b) is a great article discussing all about the functional format of `setState`.

Reply:
Before I saw it from this article, I recognized that setState works asynchronously while I'm trying to observe the handleClick method with console.log outputs. Then, I installed the officially provided react-dev extension to observe state changings. Thank you for showing me to handle these async issues. It's really hard to recognize what's going on there without seeing it from a basic sample. I may simulate it wrongly, can you evaluate it?

When I called the next sequantially code parts, it's just increment test by 1, but if I do it in a functional way, I got 3
`
this.setState({ test: this.state.test + 1});
this.setState({ test: this.state.test + 1});
this.setState({ test: this.state.test + 1});
`

Reply:
Also, it changed in the related [commit](https://github.com/btuerker/react-calculator/pull/1/commits/9aea65d6f76d5b80d6885b4c42a9837d72f0e5b1).